weatherrouting/routers/router.py

```diff
@@ -287,89 +321,24 @@ def cisopoints(i):
-
-        # Remove slow isopoints inside
-        bearing = {}
-        for x in newisopoints:
-            k = str(int(math.degrees(x.startWPLos[1])))
-
-            if k in bearing:
-                if x.nextWPDist < bearing[k].nextWPDist:
-                    bearing[k] = x
-            else:
-                bearing[k] = x
-
-        isonew = self._filterValidity(list(bearing.values()), last)
-        isonew = sorted(isonew, key=(lambda a: a.startWPLos[1]))
-        isocrone.append(isonew)
-
-        return isocrone
-
-    def _calculateIsochrones(self, t, dt, isocrone, nextwp, pointF):
-        """Calcuates isochrones based on pointF next point calculation"""
-        last = isocrone[-1]
-
-        newisopoints = []
-
         # foreach point of the iso
-        for i in range(0, len(last)):
-            p = last[i]
 
-            try:
-                (twd, tws) = self.grib.getWindAt(t, p.pos[0], p.pos[1])
-            except Exception as e:
-                raise RoutingNoWindException() from e
+        if self.getParamValue("concurrent"):
+            executor = ThreadPoolExecutor()
+            for x in executor.map(_calculateIsoPoints, range(0, len(last))):
+                newisopoints.extend(x)
 
-            for twa in range(-180, 180, 5):
-                twa = math.radians(twa)
-                twd = math.radians(twd)
-                brg = utils.reduce360(twd + twa)
-
-                # Calculate next point
-                ptoiso, speed = pointF(p.pos, tws, twa, dt, brg)
-
-                nextwpdist = utils.pointDistance(
-                    ptoiso[0], ptoiso[1], nextwp[0], nextwp[1]
-                )
-                startwplos = isocrone[0][0].lossodromic((ptoiso[0], ptoiso[1]))
-
-                if nextwpdist > p.nextWPDist:
-                    continue
-
-                # if self.pointValidity:
-                # 	if not self.pointValidity (ptoiso[0], ptoiso[1]):
-                # 		continue
-                # if self.lineValidity:
-                # 	if not self.lineValidity (ptoiso[0], ptoiso[1], p.pos[0], p.pos[1]):
-                # 		continue
-
-                newisopoints.append(
-                    IsoPoint(
-                        (ptoiso[0], ptoiso[1]),
-                        i,
-                        t,
-                        twd,
-                        tws,
-                        speed,
-                        math.degrees(brg),
-                        nextwpdist,
-                        startwplos,
-                    )
-                )
+            executor.shutdown()
+        else:
+            for i in range(0, len(last)):
+                newisopoints += _calculateIsoPoints(i)
 
         newisopoints = sorted(newisopoints, key=(lambda a: a.startWPLos[1]))
 
         # Remove slow isopoints inside
         bearing = {}
         for x in newisopoints:
-            k = str(int(math.degrees(x.startWPLos[1])))
+            k = str(int(math.degrees(x.startWPLos[1]) / subdiv))
```

Comment:
I guess that this is where the magic happen. I don't understand how you filter the iso points, if not too long can you explain?

Reply:
Can you also explain how you set concurrency?

Reply:
> I guess that this is where the magic happen. I don't understand how you filter the iso points, if not too long can you explain?

if you don't filter isopoint, you will get an enormous number of points at every step. This optional subdivision introduces the ability to reduce result resolution, so results that differs for a small amount of degrees are aggregated





> Can you also explain how you set concurrency?

as other params `routing_obj.algorithm.setParamValue("concurrent", True)`; anyway it's disabled by default since it is apparently slower